### PR TITLE
Add description of `observation_key` in `XGBoostPruningCallback`

### PR DESCRIPTION
**Pull request by smly**
## Motivation

This PR adds a description of `observation_key` in `XGBoostPruningCallback`.
Related issue: https://github.com/optuna/optuna/issues/1237.

XGBRegressor and XGBClassifier always uses a metric name like `validation_(index)-(metric)`, not `validation-(metric)`. This PR adds the description to notice this point.

## Description of the changes

This change adds the following description:

```
Note that when using the sklearn API callback,
the index number of `eval_set` must be included in the `observation_key`.
e.g., ``validation_0-error`` and ``validation_0-merror``.
```

### CODE REVIEW

File: optuna/integration/xgboost.py [hunk 1]
```diff
@@ -43,6 +43,9 @@ class XGBoostPruningCallback(object):
             ``validation-merror``. Please refer to ``eval_metric`` in
             `XGBoost reference <https://xgboost.readthedocs.io/en/latest/parameter.html>`_
             for further details.
+            Note that when using the sklearn API callback, the index number of `eval_set`
```
Comment (toshihikoyanase) [inline optuna/integration/xgboost.py:46]: I have four comments on this line.

1. How about inserting this sentence before the previous line (e.g., `validation-merror`. When using the Scikit-Learn API, the index number of ...)? IMO, it will be easier to read if the explanation of ``observation_key`` is not separated. 
2. could you replace "sklearn API callback" with "Scikit-Learn API" since the [xgboost reference](https://xgboost.readthedocs.io/en/latest/python/python_api.html#module-xgboost.sklearn) calls it "Scikit-Learn" API?
3. How about adding a link to the reference? 
4. `eval_set` needs one more pair of backticks.

image

Comment (smly) [reply optuna/integration/xgboost.py:46]: Thanks for your review. Totally agree with your suggestion of (1), (2), (4).
As for (3), it is not documented that `observation_key` is passed to the callback function in a format like `validation_0-merror`. So I don't think adding [references](https://xgboost.readthedocs.io/en/latest/python/python_api.html#module-xgboost.sklearn) is helpful. How do you think about it?

Comment (toshihikoyanase) [reply optuna/integration/xgboost.py]: Thank you for your update. 

> As for (3), it is not documented that `observation_key` is passed to the callback function in a format like `validation_0-merror`. 

That makes sense. I agree to your opinion.
